.github/workflows/code-quality.yml

```diff
@@ -41,6 +41,13 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
+      - name: Cache ccache cache
+        uses: actions/cache@v2
+        with:
+          path: /tmp/ccache
+          key: ubuntu-20.04-ccache-lint-${{ github.run_id }}
+          restore-keys: ubuntu-20.04-ccache-lint-
```

Comment:
[question] Do we not need to do this with Conan as it already namespaces its folder with hashes?

Reply:
Not quite sure what you mean. We don't need to do this with conan because in theory it never changes, unlike build artifacts which change with most PRs.

:thinking: Perhaps we should do this for conan as well, though, otherwise if we add a new package to our conanfile then it will always be reinstalled, until the cache is evicted (which mysteriously happens occasionally, see #612)

I should at least add a comment in the yml to explain what this is doing.

Reply:
Yeah, that last situation was what I was thinking. But I had noticed there are some hash prefixed dirs inside of the connan folder anyway, thought Id seen that changing when the conanfile changes... hadn't had time to look into what they were used for.

Reply:
I added a comment to the yml in 7bd9b696.

I updated the conan cache to use the same strategy as the ccache cache in 6fc03219.

Reply:
Frustratingly, the CI checks are not running yet because of a merge conflict (presumably) :roll_eyes:

Reply:
Yeah, feel free to rebase/squash ready for merge